codec-socks/src/main/java/io/netty/handler/codec/socks/SocksAuthRequestDecoder.java

```diff
@@ -62,7 +63,8 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf byteBuf, List<Object> o
         ctx.pipeline().remove(this);
     }
 
-    enum State {
+    @UnstableApi
+    public enum State {
```

Comment:
How is this exposed ? Is it because of the `state()` method ? If this is the case we should not mark it public but `protected` . Same comment for others

Reply:
Yeah, the `state` method from the super-class is `protected` and the sub-class is not `final`, hence it can in theory be extended by another sub-class which then gain access to the `state` method and its generic return type, which is this `State` enum.